packages/neon-core/src/sc/manifest/ContractPermissionDescriptor.ts

```diff
@@ -0,0 +1,48 @@
+export class ContractPermissionDescriptor {
+  public hash?: string;
+  public group?: string;
+
+  public get isHash(): boolean {
+    return !!this.hash;
+  }
+
+  public get isGroup(): boolean {
+    return !!this.group;
+  }
+
+  public get isWildcard(): boolean {
+    return !this.hash && !this.group;
+  }
+
+  private constructor(hash?: string, group?: string) {
```

Comment:
Am I correct to say a permission must be either a hash or group? Cannot be both? This constructor is misleading. I see that we can always infer from the length so there is no need for 2 params. What is a group? Need more context before I can suggest an alternative.

Reply:
> Am I correct to say a permission must be either a hash or group?

Yes.
> This constructor is misleading. 

Kind of.
> What is a group?

A group is a public key for a group of contracts. Usually for a Dapp or onchain service, more than 1 smart contract is needed, contract owner can group all Dapp related contracts for permission management.

Reply:
I think for now, its easy to just accept a single string constructor and infer the meaning since its pretty clear that the length of the string would dictate the type of permission.

Reply:
OK